src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -660,6 +657,30 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
                 }
             }
 
+            private async Task<bool> ComputeDependentCompleteAsync(Solution solution, bool isComplete, CancellationToken cancellationToken)
+            {
+                if (!isComplete)
+                {
+                    return false;
+                }
+
+                foreach (var projectReference in this.ProjectState.ProjectReferences)
```

Comment:
the other loop is not done in a way getting compilation is straightfoward, it pass down projects and at some point it create compilation. this is way simpler.